BadCode.php

```
@@ -0,0 +1,67 @@
+<?php
+// demo_code/BadCode.php - 故意写的有问题的代码
+class OrderProcessor
+{
+    public function processOrder($orderId, $userId, $items, $discount, $shippingAddress, $billingAddress, $paymentMethod, $specialInstructions) 
```

PHPMD (2.15.0)

Avoid unused parameters such as '$discount'. (Unused Code Rules)

(UnusedFormalParameter)